### PR TITLE
Polish `ServletWebRequest`

### PR DESCRIPTION
**Pull request by smartandhandsome**
This PR includes the following changes to improve code readability:

1. **Code readability**:
   - Refactored an `if` statement to enhance readability.
   - Removed an unnecessary `else` block
   
2. **Javadoc improvements**:
   - Replaced `@code` with `@link` for the `checkNotModified(String)` reference to create clickable links in the Javadoc.

### CODE REVIEW

File: spring-web/src/main/java/org/springframework/web/context/request/WebRequest.java [hunk 1]
```diff
@@ -223,7 +223,7 @@ public interface WebRequest extends RequestAttributes {
 	 * also with conditional POST/PUT/DELETE requests.
 	 * <p><strong>Note:</strong> The HTTP specification recommends
 	 * setting both ETag and Last-Modified values, but you can also
-	 * use {@code #checkNotModified(String)} or
+	 * use {@link #checkNotModified(String)} or
```
Comment (sdeleuze) [inline spring-web/src/main/java/org/springframework/web/context/request/WebRequest.java:226]: The use of `@code` is done on purpose, as a `@link` to the current method generates a warning (in IntelliJ IDEA at least), so I will keep this part as it is.

Comment (smartandhandsome) [reply spring-web/src/main/java/org/springframework/web/context/request/WebRequest.java:226]: Thank you for your feedback!!

But,,, I think this is an overloaded method and not the same one, so there’s no warning in IntelliJ.

Comment (sdeleuze) [reply spring-web/src/main/java/org/springframework/web/context/request/WebRequest.java:226]: Good catch, I checked the wrong one, I will bring back this change via an additional commit.

Comment (smartandhandsome) [reply spring-web/src/main/java/org/springframework/web/context/request/WebRequest.java:226]: Thank you!!!

**Pull request closed without merging**